### PR TITLE
Cleanup SlicePositionsAppender

### PR DESCRIPTION
**Pull request by lukasz-stec**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

improvement
> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

core query engine
> How would you describe this change to a non-technical end user or system administrator?

Code refactoring
## Related issues, pull requests, and links

In response to discussion [here](https://github.com/trinodb/trino/pull/11289#discussion_r895600436)

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/operator/output/SlicePositionsAppender.java [hunk 1]
```diff
@@ -123,21 +123,18 @@ public void appendRle(RunLengthEncodedBlock block)
         if (rlePositionCount == 0) {
             return;
         }
-        int sourcePosition = 0;
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/operator/output/SlicePositionsAppender.java:126]: nit: separate commits

File: core/trino-main/src/main/java/io/trino/operator/output/SlicePositionsAppender.java [hunk 2]
```diff
@@ -185,18 +182,18 @@ private void copyBytes(Block block, int[] lengths, int[] positions, int count, i
...
     /**
      * Copy {@code length} bytes from {@code block}, at position {@code position} to {@code count} consecutive positions in the {@link #bytes} array.
      */
-    private void duplicateBytes(Block block, int position, int count, int startOffset)
+    private void duplicateBytes(Block block, int position, int count)
     {
         int length = block.getSliceLength(position);
-        int newByteCount = count * length;
-        ensureBytesCapacity(currentOffset + newByteCount);
+        int newByteCount = toIntExact((long) count * length);
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/operator/output/SlicePositionsAppender.java:194]: nit: separate commit

**Review by sopel39 — APPROVED** [review]

**Comment by lukasz-stec** [comment]
I ran a subset of `BenchmarkPartitionedOutputOperator` jmh to confirm no regressions cc @sopel39 
```
before
Benchmark                                   (channelCount)  (enableCompression)  (nullRate)  (partitionCount)  (positionCount)   (type)  Mode  Cnt     Score    Error  Units
BenchmarkPartitionedOutputOperator.addPage               1                false           0                16             8192  VARCHAR  avgt   10  2218.089 ± 32.786  ms/op
BenchmarkPartitionedOutputOperator.addPage               1                false         0.2                16             8192  VARCHAR  avgt   10  2028.034 ± 24.532  ms/op

after
Benchmark                                   (channelCount)  (enableCompression)  (nullRate)  (partitionCount)  (positionCount)   (type)  Mode  Cnt     Score    Error  Units
BenchmarkPartitionedOutputOperator.addPage               1                false           0                16             8192  VARCHAR  avgt   10  2217.335 ± 30.768  ms/op
BenchmarkPartitionedOutputOperator.addPage               1                false         0.2                16             8192  VARCHAR  avgt   10  2035.245 ± 29.503  ms/op
```

**Comment by lukasz-stec** [comment]
@sopel39 I divided the changes in separate commits

**Pull request merged**